test/testdrive/statistics-deletion.td

```diff
@@ -0,0 +1,181 @@
+# included in the LICENSE file at the root of this repository.
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0.
+
+$ postgres-execute connection=postgres://mz_system:materialize@${testdrive.materialize-internal-sql-addr}
+ALTER SYSTEM SET enable_format_json = true
+ALTER SYSTEM SET enable_unmanaged_cluster_replicas = true
+
+# Clean up cluster manually, since testdrive does not automatically clean up
+# clusters.
+> DROP CLUSTER IF EXISTS cluster1;
+
+> CREATE CLUSTER cluster1 REPLICAS (r1 (SIZE '2'))
+
+> CREATE CONNECTION kafka_conn
+  TO KAFKA (BROKER '${testdrive.kafka-addr}');
+
+> CREATE TABLE t (a text, b text)
+> CREATE MATERIALIZED VIEW simple_view AS SELECT * from t;
+> INSERT INTO t VALUES ('key1', 'value')
+
+# Setup various sinks and sources
+> CREATE SINK sink_size FROM simple_view
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'topic-${testdrive.seed}')
+  KEY (a)
+  FORMAT JSON
+  ENVELOPE DEBEZIUM
+  WITH (SIZE = '2')
+
+> CREATE SINK sink_cluster
+  IN CLUSTER cluster1
+  FROM simple_view
+  INTO KAFKA CONNECTION kafka_conn (TOPIC 'topic-${testdrive.seed}')
+  KEY (a)
+  FORMAT JSON
+  ENVELOPE DEBEZIUM
+
+$ kafka-create-topic topic=upsert partitions=1
+$ kafka-ingest format=bytes topic=upsert key-format=bytes key-terminator=:
+one:two
+
+> CREATE SOURCE upsert_size
+  FROM KAFKA CONNECTION kafka_conn (TOPIC
+  'testdrive-upsert-${testdrive.seed}'
+  )
+  KEY FORMAT BYTES
+  VALUE FORMAT BYTES
+  ENVELOPE UPSERT
+  WITH (SIZE = '2')
+
+> CREATE SOURCE upsert_cluster
+  IN CLUSTER cluster1
+  FROM KAFKA CONNECTION kafka_conn (TOPIC
+  'testdrive-upsert-${testdrive.seed}'
+  )
+  KEY FORMAT BYTES
+  VALUE FORMAT BYTES
+  ENVELOPE UPSERT
+
+> CREATE SCHEMA subsources_size;
+> CREATE SCHEMA subsources_cluster;
+> CREATE SOURCE subsources_size.s FROM LOAD GENERATOR AUCTION FOR ALL TABLES WITH (SIZE = '2');
+> CREATE SOURCE subsources_cluster.s IN CLUSTER cluster1 FROM LOAD GENERATOR AUCTION FOR ALL TABLES;
+
+# NOTE: These queries are slow to succeed because the default metrics scraping
+# interval is 30 seconds. Here we are just ensuring metrics were populated.
+
+> SELECT s.name, SUM(u.messages_staged), SUM(u.messages_committed), SUM(u.bytes_staged) > 0, SUM(bytes_staged) = SUM(bytes_committed)
+  FROM mz_sinks s
+  JOIN mz_internal.mz_sink_statistics u ON s.id = u.id
+  WHERE s.name IN ('sink_size', 'sink_cluster')
+  GROUP BY s.name
+sink_size 1 1 true true
+sink_cluster 1 1 true true
+
+> SELECT s.name,
+  SUM(u.updates_committed) > 0
+  FROM mz_sources s
+  JOIN mz_internal.mz_source_statistics u ON s.id = u.id
+  WHERE s.name IN ('upsert_size', 'upsert_cluster', 's', 'bids')
+  GROUP BY s.id, s.name
+upsert_size true
+upsert_cluster true
+s false
+s false
+bids true
+bids true
+
+# We have to obtain these before we delete the sink.
+$ set-from-sql var=sink-size-id
+SELECT s.id
```

Comment:
I recently added `mz_internal.mz_resolve_object_name`, which would let you track this as:

```sql
SELECT id FROM mz_internal.mz_resolve_object_name('sink_size')
```

Realize you already wrote this but just so you experience less pain in the future.

Reply:
In the future I will absolutely use that!!